datalad/support/gitrepo.py

```diff
@@ -2504,6 +2506,9 @@ def _fetch_push_helper(
                         pi = info_cls._from_line(line)
                         if add_remote:
                             pi['remote'] = remote
+                        # There were errors, but Git provided hints
+                        if 'error' in pi['operations']:
+                            pi['hints'] = hints if hints is not '' else None
```

Comment:
`is` should be avoided because you're comparing a string literal.  (IIRC Python actually started warning about this in a very recent version, perhaps 3.8.)

You could use `==`, but in this case I think `hints or None` would suffice.

Reply:
Thanks. To be honest, I don't know how/why this (`hints or None`) works. What can I google to find out about this?

Reply:
Hmm, I first checked `pydoc or`.  Ignoring the BNF notation and skipping down to the "x or y" paragraph, I think that description is pretty good:

```
The expression "x or y" first evaluates *x*; if *x* is true, its value
is returned; otherwise, *y* is evaluated and the resulting value is
returned.
```

So, `x if x else y` but slightly shorter (and more easily parsed, depending on the eyes).

Searching online for "python boolean operations", I had to scroll past a few tutorials, but not too far down saw a link to [this part of the python docs][0], which says the same thing as the `pydoc or` page, perhaps more clearly.

[0]: https://docs.python.org/3.8/library/stdtypes.html#boolean-operations-and-or-not